apps/web/pages/bookings/[status].tsx

```diff
@@ -118,3 +126,51 @@ export default function Bookings() {
     </Shell>
   );
 }
+
+export const getServerSideProps = async (context: GetServerSidePropsContext) => {
+  const { req } = context;
+  const session = await getSession({ req });
+
+  if (!session?.user?.id) {
+    return {
+      redirect: {
+        permanent: false,
+        destination: "/auth/login",
+      },
+    };
+  }
+
+  const credentials = await prisma.credential.findMany({
+    where: {
+      userId: session.user.id,
+    },
+    select: {
+      id: true,
+      type: true,
+      key: true,
+      userId: true,
+      appId: true,
+    },
+  });
+
+  const integrations = getApps(credentials);
+
+  const currentUser = await prisma?.user.findFirst({
+    where: {
+      id: session.user.id,
+    },
+    select: {
+      locale: true,
+    },
+  });
+
+  const t = await getTranslation(currentUser?.locale ?? "en", "common");
+
+  const locationOptions = getLocationOptions(integrations, t);
+
+  return {
+    props: {
+      locationOptions,
```

Comment:
@emrysal and @zomars, I am fetching `locationOptions` in `getServerSideProps` now and pass it as a prop to the `EditLocationDialog`. In event-types/[type].tsx we already get `locationOptions` from `getServerSideProps`. Let me know what you think

Reply:
Ideally we'd live in a world without getServerSideProps - reason being is that it forces a reload of the entire page, much like a non-JS app. So if possible always avoid using it, and prefer doing something async.

Reply:
Agreed @emrysal although since it's a dashboard it doesn't matter that much. We should be extra careful with public pages tho. Like `sucess.tsx`